### PR TITLE
Add FAQ section

### LINKED ISSUE #13: FAQ > Question Section
Container to hold a section of FAQ questions
Screen Shot 2021-07-15 at 11 01 48 PM

### LINKED ISSUE #32: FAQ > Container (desktop + mobile)
Add all appropriate components - include dummy text as CMS connection needs to be established

Screen Shot 2021-08-04 at 6 56 38 PM
Screen Shot 2021-08-04 at 6 57 01 PM

### PR DESCRIPTION
**Pull request by kevin-zou**
## Description
Resolves #13. Resolves #32. Adding in the FAQ section! This fetches from the Firestore so all we need to do is actually populate it with data and then we're good to go. 

Currently it's hardcoded to have the Hacker and Sponsor categories of FAQs in the left side and the General FAQs in the right side, but I think that's intentional.

Desktop
image

Mobile
image

## Other considerations
- made some minor tweaks to the `FaqBox` component to make it match designs

### CODE REVIEW

**Comment by kevin-zou** [comment]
Looks like you'll have to pull this down yourself and make sure you have the right `.env` config to see the changes. Not sure why, will start a Slack thread

File: components/Faq.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            categories[faq.category] = [];
+        }
+        categories[faq.category].push(faq);
+    });
+
+    return (
+        <FaqContainer>
+            <FlexContainer>
+                {categories['Hackers'] && <FaqSection category={'Hackers'} faqs={categories['Hackers']}/>}
+                {categories['Sponsors'] && <FaqSection category={'Sponsors'} faqs={categories['Sponsors']}/>}
```
Comment (andykwan1124) [inline components/Faq.js:45]: Are we sure the question categories will be these? On the figma the categories are these 3, but on our production db the FAQ categories are something else (General, Logistics, Team & projects...)

**Review by andykwan1124 — COMMENTED** [review]
Generally looks good! Thanks for your work!!!

File: components/Faq.js (hunk 1 above)
Comment (andykwan1124) [reply components/Faq.js:45]: oh wait, those FAQs are for other hackathons. I assume we will just be creating new FAQs for `www` with the right categories. That works!

**Comment by andykwan1124** [comment]
getting some weird permission denied error when trying to start site locally... I will try to review tmrw if no one else reviewed after I get back.

**Comment by kevin-zou** [comment]
> getting some weird permission denied error when trying to start site locally... I will try to review tmrw if no one else reviewed after I get back.

Do you have a local `.env` file with the right config?

**Review by vincentchiang1 — APPROVED** [review]
awesome job! very well written code :)

File: components/Faq.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            margin-top: 24px;
+        }
+    }
+`
+
+export default function Faq({ faqs }) {
+    // Bucket each FAQ into a dictionary where the key is the category
+    var categories = {};
+    faqs.forEach((faq) => {
+        if (categories[faq.category] == undefined) {
```
Comment (panjenny0) [inline components/Faq.js:35]: Nit: I think you could do 
`if (!categories[faq.category])`

**Review by panjenny0 — APPROVED** [review]
This is really well done!! Thank you for your hard work 🥺

File: components/Faq.js (hunk 2 above)
Comment (kevin-zou) [reply components/Faq.js]: Ohhh good call! Thanks :)) added in 4d2b2e7

**Pull request merged**